WebDevStudios/wd_s
# Restructure and ignore generated assets

Closes #459 

### DESCRIPTION ###
1. `style.css` contains only the required masthead/metadata.
1. `assets/` contains two directories: `src/` that contains all of the source JS/CSS/images/fonts and `dist/` that contains the generated assets.
1. Gulp/webpack should place generated files under `dist/`
1. The theme now loads un-minified assets from `dist/` when `SCRIPT_DEBUG` is set to `true`.
1. `assets/dist` has been added to `.gitignore`.

### SCREENSHOTS ###
![screenshot](https://cl.ly/1957bd671a07/Image%2525202019-02-01%252520at%2525203.28.53%252520PM.png)

### OTHER ###

### STEPS TO VERIFY ###
1. Pull down this branch
1. Run `npm install`
1. Add `define( SCRIPT_DEBUG, true );` to `wp-config.php`
1. Run `npm run dev` to build all of the files
1. Load the theme in the browser. All of the assets should load. You should not see any 404s.
1. Now remove the `SCRIPT_DEBUG` from `wp-config.php`
1. Run `npm run build`
1. Load the theme in the browser. All assets should load with `.min` as the filename suffix. You should not see any 404s.

### DOCUMENTATION ###
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?
Yes. We'll need instructions on removing the `assets/dist` line in `.gitignore`.
